src/main/java/com/joelj/jenkins/eztemplates/jobtypes/JobsFacade.java

```diff
@@ -0,0 +1,113 @@
+package com.joelj.jenkins.eztemplates.jobtypes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.kohsuke.stapler.Ancestor;
+
+import com.google.common.base.Throwables;
+import com.joelj.jenkins.eztemplates.exclusion.Exclusions;
+
+import hudson.model.AbstractProject;
+import hudson.model.Job;
+import hudson.model.Label;
+import hudson.scm.SCM;
+import hudson.triggers.Trigger;
+import hudson.triggers.TriggerDescriptor;
+
+
+/**
+ * Facade to encapsulate the access to different types of Jenkins Jobs. Should be used when direct use of Job isn't possible and you have
+ * to deal with concrete implementations of Jobs.
+ *
+ * @see <a href="https://github.com/jenkinsci/pipeline-plugin/blob/master/DEVGUIDE.md#historical-background">pipeline's DEVGUIDE.md#historical-background</a>
+ * @author Juan Pablo Santos Rodríguez
+ */
+public class JobsFacade {
+
+    private static boolean pipelinesPluginEnabled = Exclusions.checkPlugin("workflow-job") == null;
+
+    /**
+     * Verifies if the the plugin applies to the Jenkins job type.
+     *
+     * @param jobType Jenkins job type.
+     * @return {@code true} if it is either an {@link AbstractProject} or a {@link WorkflowJob}.
+     */
+    public static boolean isPluginApplicableTo( Class< ? extends Job > jobType ) {
+        if( pipelinesPluginEnabled ) {
+            return AbstractProject.class.isAssignableFrom( jobType )
+                || WorkflowJob.class.isAssignableFrom( jobType );
+        }
+        return AbstractProject.class.isAssignableFrom( jobType );
+    }
+
+    public static List< Job > getAllTemplatableJobs() {
+        List< Job > jobs = new ArrayList<>();
+        jobs.addAll( getApplicableJobOperationsFor( AbstractProject.class, null ).getAllJobs() );
+        jobs.addAll( getApplicableJobOperationsFor( WorkflowJob.class, null ).getAllJobs() );
+        return jobs;
+    }
+
+    public static List< Job > getAllJobs( Class< ? extends Job > jobType ) {
+        List< Job > jobs = new ArrayList<>();
+        jobs.addAll( getApplicableJobOperationsFor( jobType, null ).getAllJobs() );
+        return jobs;
+    }
+
+    public static Ancestor findTemplatableAncestorFrom(Ancestor ancestor) {
+        Ancestor job = ancestor;
+        while (job != null && ( !(job.getObject() instanceof AbstractProject) ||
+                                !(job.getObject() instanceof WorkflowJob) ) ) {
+            job = job.getPrev();
+        }
+        return job;
+    }
+
+    public static Label getAssignedLabel( Job job ) {

```

Comment:
**Applies to all exclusions and is the main feedback of this review**

I am not comfortable that many encapsulated Exclusions now call out to `JobsFacade` just so that the pipeline variant can do a no-op. Is there a way we can achieve the same goal through `Exclusions.configuredExclusions` (i.e. do not return the ScmExclusion if the job is a pipeline)?